client/src/components/geneExpression/menus/addGeneToGenesetDialogue.js

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import _ from "lodash";
+import { connect } from "react-redux";
+import AnnoDialog from "../../annoDialog";
+import LabelInput from "../../labelInput";
+
+@connect((state) => ({
+  genesetsUI: state.genesetsUI,
+}))
+class AddGeneToGenesetDialogue extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      genesToAdd: "",
+    };
+  }
+
+  disableAddGeneMode = () => {
+    const { dispatch } = this.props;
+    dispatch({
+      type: "geneset: disable add new genes mode",
+    });
+  };
+
+  handleAddGeneToGeneSet = (e) => {
+    const { geneset, dispatch } = this.props;
+    const { genesToAdd } = this.state;
+
+    const genesTmpHardcodedFormat = [];
+
+    const genesArrayFromString = _.pull(_.uniq(genesToAdd.split(/[ ,]+/)), "");
```

Comment:
this little bit of magic is repeated in `createGenesetDialogue`.  Suggest pulling it out into a helper utility function given its generic behavior (ie, split string by comma or space, return unique non-empty strings)